demos/question-answering/README.md

```diff
@@ -23,44 +38,83 @@ To complete the tutorial follow the steps below:
+- Connect to the endpoint and submit your questions.
+
+![application-ui](images/application-ui.png)
 
 ## How It Works
 
-This project utilizes a [HuggingFace embedding model](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2) to map the sentences and paragraphs of a corpus of private documents to a multi-dimensional dense vector space. This enables the use of advanced techniques like semantic search, which can help users find relevant information quickly and accurately. The resulting vectors are stored and indexed in [Chroma](https://www.trychroma.com/), an AI-native, open-source embedding database.
+This project utilizes a HuggingFace [embedding model](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2) [1]
+to map the sentences and paragraphs of a corpus of private documents to a multi-dimensional dense vector space [2]. This
+enables the use of advanced techniques like semantic search, which can help users find relevant information quickly and
+accurately. The resulting vectors are stored and indexed in a local instance of [Chroma](https://www.trychroma.com/), an
+AI-native, open-source embedding database [3].
 
-Once a question is provided, the application embeds it into the same vector space using the same transformer model. It then retrieves the `4` most relevant documents by applying a certrain algorithm (by default `kNN`). Finally, the application passes the user's question and the retrieved context to an LLM, which can answer the question in a human-like way. By leveraging advanced techniques like vector embeddings and LLMs, the application provides a powerful and intuitive way to search and analyze private documents.
+Once a question is provided, the application embeds it into the same vector space using the same transformer model. It
+then retrieves the `4` most relevant documents by default, applying a certrain algorithm (by default `kNN`) [4].
+Finally, the application passes the user's question and the retrieved context to an LLM, which can answer the question
+in a human-like way. For the LLM model, the tutorial supports any GPT4ALL [5] supported architecture. By default, we use
+the `orca-mini-3b.ggmlv3.q4_0` model.
 
-## Usage
+By leveraging advanced techniques like vector embeddings and LLMs, the application provides a powerful and intuitive way
+to search and analyze private documents.
 
-KServe allows you to create an inference service using a custom predictor. Since there's no out-of-the-box support for LLMs yet, you need to create the images that KServe will use to serve the Vector Store and the LLM. As such, you need to create three images and push them to a registry you have access:
+In practice, [KServe](https://kserve.github.io/website/0.11/) [6] allows you to create an inference service using a
+custom predictor. Since there's no out-of-the-box support for LLMs yet, you need to create the images that KServe will
```

Comment:
I'd recommend "We need to create" in place of "Since there's no... you need to create"

Reply:
"We need to create the three images in a registry that KServe will use to serve the Vector Store and the LLM:", and remove "As such, ... you have access"

Reply:
"We need to create the three images in a registry that KServe will use to serve the Vector Store and the LLM. Here are the Dockerfile to build the associated images:

1. Vector Store: `dockerfiles/vectorstore`"
2. LLM Transformer: `dockerfiles/transformer`
3. LLM Model: `dockerfiles/llm`

Reply:
replace the `dockerfiles` with `apps` if you agree on changing the folder name.

Reply:
I re-wrote it like this:

```
Using [KServe](https://kserve.github.io/website/0.11/) [6] you can set up an Inference Service with custom components.
To do this, you need to build and push three images which KServe will use to serve both the Vector Store and the LLM.
Below are the application directories containing the respective Dockerfiles to build these images:

- Vector Store: [`dockerfiles/vectorstore`](dockerfiles/vectorstore)
- LLM Transformer: [`dockerfiles/transformer`](dockerfiles/transformer)
- LLM Model: [`dockerfiles/llm`](dockerfiles/llm)
```